src/lap_stats/lap_stats.c

```diff
@@ -129,15 +129,22 @@ static void end_lap_timing(const GpsSnapshot *gpsSnapshot)
     g_lapStartTimestamp = -1;
 }
 
-static void update_distance(const GpsSnapshot *gpsSnapshot)
+static void update_distance(const GpsSnapshot *gps_ss)
 {
-    const GeoPoint prev = gpsSnapshot->previousPoint;
-    const GeoPoint curr = gpsSnapshot->sample.point;
+        const float speed_avg =
+                (gps_ss->sample.speed + gps_ss->previous_speed) / 2;
 
-    if (!isValidPoint(&prev) || !isValidPoint(&curr))
-        return;
+        if (speed_avg < 1)
```

Comment:
Understood. I would recommend documenting this, and removing the magic number so the intention is obvious.  Thanks!